iroha/src/queue.rs

```diff
@@ -54,19 +92,15 @@ impl Queue {
         &self,
         tx: VersionedAcceptedTransaction,
         wsv: &WorldStateView<W>,
-    ) -> Result<(), VersionedAcceptedTransaction> {
-        if tx.is_expired(self.ttl) {
-            iroha_logger::warn!("Transaction expired");
-            return Err(tx);
-        }
-        if tx.is_in_blockchain(wsv) {
-            iroha_logger::warn!("Transaction is already applied");
-            return Err(tx);
+    ) -> Result<(), Error> {
+        // Should always check transactions to prevent eternal gossiping of wrong txs.
+        if let Err(error) = self.check_tx(&tx, wsv) {
+            iroha_logger::warn!("{}", error);
```

Comment:
It is a bad idea to do both logging here and return error. It is better to add choose whether to report on caller side. For cases like transaction forwarding it makes sense to not report `InBlockchain` error. You should check out pr of mine. Some of the logs were improved there.

Reply:
Sure, I'll check. I had this in mind just did not plan to change a lot of code in this PR. But I guess I will add this logic.